helpers/helpers.py

```diff
@@ -20,8 +20,78 @@
 import config.config as cfg
 import discord
 import requests
+import os
 from typing import Optional, Dict, Any, List, Tuple
 from logger.logger import logger
+from web3 import Web3
+
+
+def fetch_XP_Tokens():
+    # Fetch RPC URLs from environment variables
+    primary_rpc = os.getenv("PRIMARY_RPC")
+    secondary_rpc = os.getenv("SECONDARY_RPC")
+
+    if not primary_rpc or not secondary_rpc:
+        logger.error("RPC URLs not set in environment variables.")
+        return
+
+    web3 = Web3(Web3.HTTPProvider(primary_rpc))
+
+    # Check if we are connected to the node
+    if not web3.is_connected():
+        logger.error("Failed to connect to PRIMARY_RPC, let's try the SECONDARY_RPC")
+        web3 = Web3(Web3.HTTPProvider(secondary_rpc))
+        if not web3.is_connected():
+            logger.error("Failed to connect to SECONDARY_RPC, shutting down")
+            exit(1)
+
+    token_abi = [
+        {
+            "constant": True,
+            "inputs": [],
+            "name": "totalSupply",
+            "outputs": [{"name": "", "type": "uint256"}],
+            "type": "function",
+        }
+    ]
+
+    token_addresses = [
+        "0x206d247F61cb82B9711318381cDb7Bc5039d2A2c",
+        "0x4cd06ada7d8564830018000d784c69bd542b1e6a",
+        "0x57d3a929fdc4faf1b35e7092d9dee7af097afb6a",
+    ]
+
+    # Initialize total supply counter
+    total_supply_sum = 0
+
+    # Loop over each token address and fetch the total supply
+    for address in token_addresses:
+        try:
+            # Convert to checksum address
+            checksum_address = Web3.to_checksum_address(address)
+            token_contract = web3.eth.contract(address=checksum_address, abi=token_abi)
+            total_supply = token_contract.functions.totalSupply().call()
+            total_supply_in_tokens = int(
+                web3.from_wei(total_supply, "ether")
+            )  # Convert to integer
+            if total_supply_in_tokens is not None:
+                logger.info(
+                    f"The total supply of the token at address {address} is {total_supply_in_tokens} tokens."
+                )
+                total_supply_sum += total_supply_in_tokens
+        except Exception as e:
+            logger.error(
+                f"An error occurred while fetching the total supply for address {address}: {str(e)}"
+            )
+
+    # Calculate 25% of the total supply
+    total_supply_25_percent = total_supply_sum * 0.25
```

Comment:
to avoid decimals, you can do ((total_supply_sum * 100) / 25)

Reply:
> to avoid decimals, you can do ((total_supply_sum * 100) / 25)

Will do